src/main/java/cpw/mods/modlauncher/LaunchPluginHandler.java

```diff
@@ -62,11 +63,21 @@ public Optional<ILaunchPluginService> get(final String name) {
         return Optional.ofNullable(plugins.get(name));
     }
 
-    public EnumMap<ILaunchPluginService.Phase, List<ILaunchPluginService>> computeLaunchPluginTransformerSet(final Type className, final boolean isEmpty, final String reason, final TransformerAuditTrail auditTrail) {
+    public EnumMap<ILaunchPluginService.Phase, List<ILaunchPluginService>> computeLaunchPluginTransformerSet(final Type className, final byte[] inputClass, final String reason, final TransformerAuditTrail auditTrail) {
         Set<ILaunchPluginService> uniqueValues = new HashSet<>();
         final EnumMap<ILaunchPluginService.Phase, List<ILaunchPluginService>> phaseObjectEnumMap = new EnumMap<>(ILaunchPluginService.Phase.class);
         for (ILaunchPluginService plugin : plugins.values()) {
-            for (ILaunchPluginService.Phase ph : plugin.handlesClass(className, isEmpty, reason)) {
+            // Check if the plugin handles classes of this name at all
+            var phaseSet = plugin.handlesClass(className, inputClass.length == 0, reason);
+            if (phaseSet.isEmpty()) {
+                continue;
+            }
+            // Filter out classes that don't match the constants filter
+            if (!ClassConstantPoolParser.constantPoolMatches(plugin.constantsFilter(className, reason), inputClass)) {
```

Comment:
We need a way to opt-out of this behavior in case it causes issues (for example due to forgetting to update the constant pool parser). Probably via an env variable or ML arg.